mautic/mautic
# Add Endpoint field to S3 plugin

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

The endpoint field will allow the plugin to be used with other S3-compliant block storage solutions like DigitalOcean Spaces.

#### Steps to test this PR:

1. Create an S3-compliant bucket on platforms like DigitalOcean Spaces
2. Generate the Spaces API token
3. On the mautic website, go to plugins and click on the S3 plugin
4. Add the relevant credentials including the endpoint found on the Spaces page
5. Under assets, you should be able to access the remote files on Spaces
<img width="604" alt="image" src="https://user-images.githubusercontent.com/2114712/200844330-7b675034-da6f-4b39-9d70-7e4b5ed8ee9a.png">
